src/aiohabit/utils.py

```diff
@@ -133,6 +133,23 @@ def print_obj(obj):
     print(object2json(obj))
 
 
+def get_host_from_metadata(metadata, name):
+    """
+    Get host definition from job metadata base on name.
+
+    Returns:
+    (host, domain)
+    """
+    domains = metadata.get("domains", [])
+    match = [
+        (host, domain)
+        for domain in domains
+        for host in domain.get("hosts", [])
+        if host["name"] == name
+    ]
```

Comment:
Final comment: I think we should avoid list comprehensions in complex cases like that, this is much easier to read and debug:
```python
for domain in domains:
    for host in domain.get("hosts", []): 
        if host["name"] == name:
            return host, domain
````
It even makes clear that if no host with `name` is found then only 1 variable will be returned ("`None`" instead of "`None, None`"), but we don't need to handle this right now and it will fail explicitly in the future.

Reply:
You are right, this code is better and more readable. Changed.